File: program_data/github_java_sort_function/3/88.java
package DFS;

import java.util.Iterator;
import java.util.LinkedList;
import java.util.Stack;

class GraphTopologicalSorting
{
    private int V;   
    private LinkedList<Integer> adj[]; 
 
    
    GraphTopologicalSorting(int v)
    {
        V = v;
        adj = new LinkedList[v];
        for (int i=0; i<v; ++i)
            adj[i] = new LinkedList();
    }
 
    
    void addEdge(int v,int w) { adj[v].add(w); }
 
    
    void topologicalSortUtil(int v, boolean visited[],
                             Stack stack)
    {
        
        visited[v] = true;
        Integer i;
 
        
        
        Iterator<Integer> it = adj[v].iterator();
        while (it.hasNext())
        {
            i = it.next();
            if (!visited[i])
                topologicalSortUtil(i, visited, stack);
        }
 
        
        stack.push(new Integer(v));
    }
 
    
    
    void topologicalSort()
    {
        Stack stack = new Stack();
 
        
        boolean visited[] = new boolean[V];
        for (int i = 0; i < V; i++)
            visited[i] = false;
 
        
        
        
        for (int i = 0; i < V; i++)
            if (visited[i] == false)
                topologicalSortUtil(i, visited, stack);
 
        
        while (stack.empty()==false)
            System.out.print(stack.pop() + " ");
    }
 

}

File: program_data/PKU_raw/60/88.c
void main()
{
	int n,m,k,i,j,r,c,q=0,p=0;
	int a[2000];
	scanf("%d",&n);
	for(i=3;i<=n;i=i+2)
	{
		c=1;
		for(j=2;j<=i/2;j++)
		{
			r=i%j;
			if(r==0) {c=0;break;}
		}
		if(c==1)
		{
			a[p]=i;
			p++;
		}
	}
	for(i=0;a[i];i++)
	{
		if(a[i+1]-a[i]==2)
		{
			printf("%d %d\n",a[i],a[i+1]);
			q++;
		}

	}
	if(q==0)printf("empty");

}